jupyterlab_caip_optimizer/src/components/suggest_trials/create_trial.tsx

```diff
@@ -0,0 +1,241 @@
+    switch (spec.type) {
+      case 'CATEGORICAL':
+        return (
+          <SelectionInput
+            key={name}
+            label={label}
+            input={values[name]}
+            onChange={value => onChange(name, value)}
+            // TODO: fix typing
+            validValues={(spec as any).categoricalValueSpec.values as string[]}
+          />
+        );
+      case 'DISCRETE':
+        return (
+          <SelectionInput
+            key={name}
+            label={label}
+            input={values[name]}
+            onChange={value => onChange(name, value)}
+            // TODO: fix typing
+            validValues={((spec as any).discreteValueSpec
+              .values as number[]).map(value => value.toString(10))}
+          />
+        );
+      case 'DOUBLE':
+      case 'INTEGER':
+        return (
+          <NumberInput
+            key={name}
+            label={label}
+            input={values[name]}
+            onChange={value => onChange(name, value)}
+          />
+        );
+      default:
+        return null;
+    }
+  });
+}
+
+interface Props {
+  studyName: string;
+  studyConfig: StudyConfig;
+  open?: boolean;
+  onClose: () => void;
+}
+
+export const CreateTrial: React.FC<Props> = ({
+  studyName,
+  studyConfig,
+  open = false,
+  onClose,
+}) => {
+  const dispatch = useDispatch();
+  const [metrics, setMetrics] = React.useState<MetricsInputs>(() =>
+    metricSpecsToMetrics(studyConfig.metrics)
+  );
+  const [loading, setLoading] = React.useState(false);
+  // Creates parameter default values of '' and setups input error handling
+  // using the custom hook
+  const [map, setInput] = useErrorStateMap(
+    parameterSpecToInputsValues(studyConfig.parameters),
+    parameterSpecToValidateInput(studyConfig.parameters)
+  );
+
+  const handleClose = () => {
+    if (!loading) {
+      setMetrics(clearMetrics(metrics));
+      onClose();
+    }
+  };
+
+  const handleSubmit = async () => {
+    setLoading(true);
+    dispatch(
+      createTrial({
+        studyName,
+        trial: {
+          state: State.COMPLETED,
+          parameters: inputValuesToParameterList(
+            errorStateMapToValueObject(map),
+            studyConfig.parameters
+          ),
+          finalMeasurement: metricsToMeasurement(metrics),
+          measurements: [],
+        },
+      })
+    );
+    setLoading(false);
+    setMetrics(clearMetrics(metrics));
+    onClose();
+  };
+
+  return (
+    <Dialog
+      open={open}
+      onClose={handleClose}
+      style={{ display: 'relative' }}
+      data-testid="createTrialDialog"
+    >
+      <DialogTitle>
+        Create Custom Trial for "{prettifyStudyName(studyName)}" Study
+      </DialogTitle>
+      <DialogContent>
+        <DialogContentText>Add custom trial data.</DialogContentText>
+        {/* Parameter inputs */}
+        <Typography component="h3">Parameters</Typography>
+        {parameterSpecToInputs(map, setInput, studyConfig.parameters)}
```

Comment:
In the dialog screenshot, “integer-value” is the only one in kebab case. Is this something grabbed from the API?

Reply:
Yes, this was pulled from the study configuration, aka a user-created value ([reference](https://github.com/GoogleCloudPlatform/jupyter-extensions/pull/140/files/1532243e8634ac47c0e3178c0b8e8e10b88472a9#diff-3ad7f71e6ff2e84669ca92678dd7e7dbR105)).

Reply:
Ah gotcha